modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb

```diff
@@ -109,21 +110,40 @@ def individual_page1_signatures(power_of_attorney, first_name, last_name)
         ]
       end
 
-      def individual_page2_signatures(power_of_attorney, first_name, last_name)
+      def individual_page2_signatures(power_of_attorney, rep_first_name, rep_last_name)
+        first_name, last_name = veteran_or_claimant_signature(power_of_attorney)
         [
           {
-            'signature' => "#{power_of_attorney.auth_headers['va_eauth_firstName']} " \
-                           "#{power_of_attorney.auth_headers['va_eauth_lastName']} - signed via api.va.gov",
+            'signature' => "#{first_name} " \
+                           "#{last_name} - signed via api.va.gov",
             'x' => 35,
             'y' => 306
           },
           {
-            'signature' => "#{first_name} #{last_name} - signed via api.va.gov",
+            'signature' => "#{rep_first_name} #{rep_last_name} - signed via api.va.gov",
             'x' => 35,
             'y' => 200
           }
         ]
       end
+
+      def veteran_or_claimant_signature(power_of_attorney)
+        claimant_icn = power_of_attorney.form_data['claimant']['claimantId']
+        if claimant_icn.present?
+          user_profile = mpi_service.find_profile_by_identifier(identifier: claimant_icn,
+                                                                identifier_type: MPI::Constants::ICN)
```

Comment:
Maybe we should check to see if the ICN is valid, or add a catch if it is not valid.
![Screenshot 2024-03-14 at 1 21 28 PM](https://github.com/department-of-veterans-affairs/vets-api/assets/25069483/0dc3c7a3-e611-4fc8-9988-905a6d9ac3a0)
![Screenshot 2024-03-14 at 1 20 43 PM](https://github.com/department-of-veterans-affairs/vets-api/assets/25069483/54fb437b-a90d-4564-880b-9214771ad9b3)

Reply:
We're now checking the ICN up front in validation, so this should be all set.